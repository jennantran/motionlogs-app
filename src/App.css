* {
    box-sizing: border-box;
  }

body {
    background-image: url('./images/barbellshadow.jpg');
    background-color: #cccccc;
   }