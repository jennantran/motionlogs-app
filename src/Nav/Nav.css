.nav{
    display: flex;
    flex-direction: row;
    align-items: center;
    }

.appName{
    padding-left: 20px;
    float: left;
    font-family: 'Slab';
}

.navLinks{
    float:right;
}

ul{
    list-style-type: none;
}

li{
    display:inline;
    margin-right: 15px;
}

a{
    text-decoration: none;
    color:#cc0000;
    font-family: 'sans-serif';
  
}
a:hover {
    text-decoration: underline;
  }


@media screen and (min-width:800px) {

a{
    font-size: 20px; 
}
}