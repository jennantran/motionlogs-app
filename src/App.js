import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import Nav from './Nav/Nav';
import './App.css';
import HomePage from './HomePage/HomePage';
import Login from './Login/Login';
import Workouts from './Workouts/Workouts';
import AddLogs from './AddLogs/AddLogs';
import Logs from './Logs/Logs';
import MotionsContext from './MotionsContext';
import SignUp from './SignUp/SignUp';

// const logs = [
//   {
//     id: '0',
//     set: '1',
//     rep: '5',
//     weight: '10'
//   },
// ]

class App extends Component {
  state = {
    logs: [],
    user_id: '1', 
    addLog: this.addLog,
    error: null,
    log_id:'1',
  };

  addLog = log => {
    this.setState({
        logs: [...this.state.logs, log]
    })
    console.log(this.state);
    console.log(this.state.logs);
  }

  setLog = logs => {
    this.setState({
      logs,
      error: null,
    })
  }

  deleteLog = logId => {
    console.log('delete log enter');
    this.setState({
     logs: this.state.logs.filter(log => log.id !== logId)
    });
  }
  render(){
    const contextValue = {
      logs: this.state.logs,
      addLog: this.addLog,
      deleteLog: this.deleteLog,
      user_id: this.state.user_id,
      log_id: this.state.log_id

    }
    return (
      <div className='app'>
        <MotionsContext.Provider value = {contextValue}>
          <nav><Nav></Nav></nav>
          <main className='App'>
              <Route exact path='/' component={HomePage} />
              <Route path='/login' component={Login} />
              <Route path='/workouts' component={Workouts} />
              <Route path='/addlogs' component={AddLogs}></Route>
              <Route path='/logs' component={Logs} />  
              <Route path='/signup' component={SignUp} />  
          </main>
          <footer></footer>
        </MotionsContext.Provider>
     </div>
    );
  }
}

export default App;