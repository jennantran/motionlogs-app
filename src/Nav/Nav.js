import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import MotionsContext from '../MotionsContext';
import TokenService from '../services/token-service';
import './Nav.css';

class Nav extends Component{
    static contextType = MotionsContext;
        constructor(props){
        super(props);
        this.state = {
            authToken: undefined,
        }
    }
    handleLogoutClick = () => {
        TokenService.clearAuthToken();
        this.setState({authToken: undefined});
        this.context.clearLogs();    
    }
  
    renderLogoutLink() {
      return (
          <li className='loggedIn'>
            <Link
              onClick={this.handleLogoutClick}
              to='/login'>
              Logout
            </Link>
          </li> 
        )
    }
  
    renderLoginLink() {
      return (
        <li className='notLogged-in'>
          <Link
            to='/login'>
            Log in
          </Link>
        </li>
      )
    }
    render(){
        return(
            <section className='nav'>
                <div className='appName'>
                     <Link to='/'><h2>Motion Logs</h2></Link>
                 </div>
                <div className='navLinks'>
                     <ul>
                         <li><Link to='/workouts'>Workouts</Link></li>
                         {TokenService.hasAuthToken()
                          ? <li><Link id='logs' to='/logs'>Logs</Link></li>
                          : null}
                          {TokenService.hasAuthToken()
                          ? this.renderLogoutLink()
                          : this.renderLoginLink()}
                     </ul>
                </div>
            </section>
        );
    }
}

export default Nav;