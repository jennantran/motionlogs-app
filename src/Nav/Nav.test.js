import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from  '../App';
import Nav from '../Nav/Nav';

it('renders Nav component without crashing', () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <App>
         <Nav></Nav>
      </App>
    </BrowserRouter>,
    div
  );
  //clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the workout UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
            <App>
                <Nav></Nav>
            </App>
        </BrowserRouter>,
      )
      .toJSON();
  
    expect(tree).toMatchSnapshot();
  });