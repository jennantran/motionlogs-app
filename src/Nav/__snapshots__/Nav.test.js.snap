// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the workout UI as expected 1`] = `
<div
  className="app"
>
  <nav>
    <section
      className="nav"
    >
      <div
        className="appName"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <h2>
            Motion Logs
          </h2>
        </a>
      </div>
      <div
        className="navLinks"
      >
        <ul>
          <li>
            <a
              href="/workouts"
              onClick={[Function]}
            >
              Workouts
            </a>
          </li>
          <li
            className="notLogged-in"
          >
            <a
              href="/login"
              onClick={[Function]}
            >
              Log in
            </a>
          </li>
        </ul>
      </div>
    </section>
  </nav>
  <main
    className="App"
  >
    <div>
      <section
        className="homepage"
      >
        <h1
          className="homeHeader"
        >
          Make your fitness journey EASIER!
        </h1>
        <p>
          Track your progress.
        </p>
        <button
          className="homeButton"
          onClick={[Function]}
        >
          Join now
        </button>
      </section>
      <section
        className="homepgDetails"
      >
        <div
          className="details design"
        >
          <h2>
            Simple.
          </h2>
          <p>
            Log your reps and weights and add the workout name.
          </p>
        </div>
        <div
          className="details learn"
        >
          <h2>
            Track.
          </h2>
          <p>
            See the amount of calories expended from your workout using the input calculator.
          </p>
        </div>
        <div
          className="details accountability"
        >
          <h2>
            Accountability.
          </h2>
          <p>
            As your numbers go up, motivation will come.
          </p>
        </div>
      </section>
    </div>
  </main>
</div>
`;
